diff --git a/file_hello1/debug/file_hello1 b/file_hello1/debug/file_hello1
index bbab98a..dae740e 100755
Binary files a/file_hello1/debug/file_hello1 and b/file_hello1/debug/file_hello1 differ
diff --git a/file_hello1/exec/file_hello1 b/file_hello1/exec/file_hello1
index c79eea7..ae0d81a 100755
Binary files a/file_hello1/exec/file_hello1 and b/file_hello1/exec/file_hello1 differ
diff --git a/file_hello1/exec/file_hello1.dSYM/Contents/Resources/DWARF/file_hello1 b/file_hello1/exec/file_hello1.dSYM/Contents/Resources/DWARF/file_hello1
index b50d3e4..e3c1854 100644
Binary files a/file_hello1/exec/file_hello1.dSYM/Contents/Resources/DWARF/file_hello1 and b/file_hello1/exec/file_hello1.dSYM/Contents/Resources/DWARF/file_hello1 differ
diff --git a/file_hello1/file_hello1.xcodeproj/project.xcworkspace/xcuserdata/daniel.xcuserdatad/UserInterfaceState.xcuserstate b/file_hello1/file_hello1.xcodeproj/project.xcworkspace/xcuserdata/daniel.xcuserdatad/UserInterfaceState.xcuserstate
index bf86ad6..91b92ad 100644
Binary files a/file_hello1/file_hello1.xcodeproj/project.xcworkspace/xcuserdata/daniel.xcuserdatad/UserInterfaceState.xcuserstate and b/file_hello1/file_hello1.xcodeproj/project.xcworkspace/xcuserdata/daniel.xcuserdatad/UserInterfaceState.xcuserstate differ
diff --git a/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist b/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
index e174021..834789a 100644
--- a/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
+++ b/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
@@ -10,11 +10,11 @@
             ignoreCount = "0"
             continueAfterRunningActions = "No"
             filePath = "file_hello1/main.m"
-            timestampString = "521583714.736497"
+            timestampString = "522258123.186767"
             startingColumnNumber = "9223372036854775807"
             endingColumnNumber = "9223372036854775807"
-            startingLineNumber = "53"
-            endingLineNumber = "53"
+            startingLineNumber = "55"
+            endingLineNumber = "55"
             landmarkName = "@interface Location"
             landmarkType = "2">
          </BreakpointContent>
@@ -26,11 +26,11 @@
             ignoreCount = "0"
             continueAfterRunningActions = "No"
             filePath = "file_hello1/main.m"
-            timestampString = "521586555.357793"
+            timestampString = "522258123.186767"
             startingColumnNumber = "9223372036854775807"
             endingColumnNumber = "9223372036854775807"
-            startingLineNumber = "50"
-            endingLineNumber = "50"
+            startingLineNumber = "52"
+            endingLineNumber = "52"
             landmarkName = "-ShowAllData"
             landmarkType = "4">
          </BreakpointContent>
@@ -38,17 +38,33 @@
       <BreakpointProxy
          BreakpointExtensionID = "Xcode.Breakpoint.FileBreakpoint">
          <BreakpointContent
-            shouldBeEnabled = "Yes"
+            shouldBeEnabled = "No"
+            ignoreCount = "0"
+            continueAfterRunningActions = "No"
+            filePath = "file_hello1/main.m"
+            timestampString = "522259448.112328"
+            startingColumnNumber = "9223372036854775807"
+            endingColumnNumber = "9223372036854775807"
+            startingLineNumber = "278"
+            endingLineNumber = "278"
+            landmarkName = "main()"
+            landmarkType = "7">
+         </BreakpointContent>
+      </BreakpointProxy>
+      <BreakpointProxy
+         BreakpointExtensionID = "Xcode.Breakpoint.FileBreakpoint">
+         <BreakpointContent
+            shouldBeEnabled = "No"
             ignoreCount = "0"
             continueAfterRunningActions = "No"
             filePath = "file_hello1/main.m"
-            timestampString = "521888628.488892"
+            timestampString = "522259448.112328"
             startingColumnNumber = "9223372036854775807"
             endingColumnNumber = "9223372036854775807"
-            startingLineNumber = "118"
-            endingLineNumber = "118"
-            landmarkName = "-ShowDescription"
-            landmarkType = "5">
+            startingLineNumber = "280"
+            endingLineNumber = "280"
+            landmarkName = "main()"
+            landmarkType = "7">
          </BreakpointContent>
       </BreakpointProxy>
       <BreakpointProxy
@@ -58,13 +74,29 @@
             ignoreCount = "0"
             continueAfterRunningActions = "No"
             filePath = "file_hello1/main.m"
-            timestampString = "521889603.333267"
+            timestampString = "522259448.112328"
+            startingColumnNumber = "9223372036854775807"
+            endingColumnNumber = "9223372036854775807"
+            startingLineNumber = "284"
+            endingLineNumber = "284"
+            landmarkName = "main()"
+            landmarkType = "7">
+         </BreakpointContent>
+      </BreakpointProxy>
+      <BreakpointProxy
+         BreakpointExtensionID = "Xcode.Breakpoint.FileBreakpoint">
+         <BreakpointContent
+            shouldBeEnabled = "No"
+            ignoreCount = "0"
+            continueAfterRunningActions = "No"
+            filePath = "file_hello1/main.m"
+            timestampString = "522260151.263425"
             startingColumnNumber = "9223372036854775807"
             endingColumnNumber = "9223372036854775807"
-            startingLineNumber = "132"
-            endingLineNumber = "132"
-            landmarkName = "-ShowDescription"
-            landmarkType = "5">
+            startingLineNumber = "288"
+            endingLineNumber = "288"
+            landmarkName = "main()"
+            landmarkType = "7">
          </BreakpointContent>
       </BreakpointProxy>
    </Breakpoints>
diff --git a/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcschemes/file_hello1.xcscheme b/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcschemes/file_hello1.xcscheme
index 13040dc..f7cb144 100644
--- a/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcschemes/file_hello1.xcscheme
+++ b/file_hello1/file_hello1.xcodeproj/xcuserdata/daniel.xcuserdatad/xcschemes/file_hello1.xcscheme
@@ -42,7 +42,7 @@
       </AdditionalOptions>
    </TestAction>
    <LaunchAction
-      buildConfiguration = "Debug"
+      buildConfiguration = "Release"
       selectedDebuggerIdentifier = "Xcode.DebuggerFoundation.Debugger.LLDB"
       selectedLauncherIdentifier = "Xcode.DebuggerFoundation.Launcher.LLDB"
       launchStyle = "0"
diff --git a/file_hello1/file_hello1/main.m b/file_hello1/file_hello1/main.m
index bce3134..274befd 100644
--- a/file_hello1/file_hello1/main.m
+++ b/file_hello1/file_hello1/main.m
@@ -11,7 +11,7 @@
 //#import <Foundation/NSFileHandle.h>
 //#import <Foundation/NSScanner.h>
 #include "stdio.h"
-
+//#include "st
 
 NSString *fileName;
 
@@ -19,7 +19,7 @@
 NSNumber *nr;
 
 
-#define  EXITS_NUMBER_IN_FILES   3
+#define  EXITS_NUMBER_IN_FILES   @"3"
 
 /*
  this classs is for 
@@ -39,7 +39,7 @@ @interface Location : NSObject
 
 @property NSNumber *location_number;
 @property NSArray *AllExits;
-@property NSArray *AllData;
+@property NSString *AllData;
 +(void)initialize;
 +(void)sayHello;
 
@@ -51,7 +51,7 @@ -(void)WriteOtherWord;
 -(NSNumber *)addTwoValues:(uint16_t)val1 withNumber:(uint16_t)val2;
 -(void)ShowAllData;
 //:(NSArray *)data;
-
+-(NSString *)WaitForCommand;
 @end
 
 
@@ -100,63 +100,63 @@ -(void)ShowDescription
     NSString *str1;
     NSInteger nr_int;
     int8_t a=0;
+    NSLog(@" I am searing for %d ",(int)nr_searched);
+    theScanner = [NSScanner scannerWithString:_AllData ];
+    NSScanner *minorScanner;
     
-    for(NSString *line in _AllData )
+//    NSLog( @" a = %d ", a);
+
+    while( [theScanner isAtEnd] == NO)
     {
-        //}
-        theScanner = [NSScanner scannerWithString:line ];
-        while( [theScanner isAtEnd] == NO)
+        [theScanner scanUpToString:@"\n" intoString:&str1];
+        
+    
+        minorScanner = [NSScanner scannerWithString:str1];
+        //if information delimiter
+        if ( ([minorScanner scanString:@"-1" intoString:NULL]) && ( a != 10) )
         {
-            if ([theScanner scanString:@"-1" intoString:NULL])
+            a++;
+//            NSLog( @" a = %d ", a);
+            NSLog(@"%@",str1);
+            break;
+            [theScanner scanUpToString:@"\n" intoString:&str1];
+            minorScanner = [NSScanner scannerWithString:str1];
+            
+            
+            if ([minorScanner scanString:EXITS_NUMBER_IN_FILES  intoString:&str1])
             {
-                //                NSLog(@" Now next set of infos");
-                if (a==1)
-                {
-                    a = 2;
-                    
-                }
-                else if (a==0)
-                {
-                    continue;
-                }
+                
+                NSLog(@" I have found exits ");
+                NSLog(@"%@",str1);
+                a = 10 ;
                 
             }
-            if ( ([theScanner  scanInteger:&nr_int]) && ([ theScanner scanLocation]<4))
+        }
+
+        
+        if ( ([minorScanner  scanInteger:&nr_int]) && ([ minorScanner scanLocation]<4) && (0 == a))
+        {
+            
+            if (nr_int  == nr_searched)
             {
-                if (nr_int  == nr_searched)
-                {
-                    //NSLog(@" I have find  %@ in %@ ",searchedString, str1);
-                    
-                    //NSLog(@" comparing %d ",(int)(nr_int == nr_searched));
-                    //[minorScanner scanUpToString:searchedString intoString:&str1];
-                    [theScanner scanUpToString:@"\n" intoString:&str1];
-                    NSLog(str1);
-                    a = 1;
-                    
-                    
-                }
-                else
-                {                    }
-                //if ([theScanner scanString:searchedString intoString:NULL])
-                //   {
+                //NSLog(@" I have find  %@ in %@ ",searchedString, str1);
                 
-                // }
+                //NSLog(@" comparing %d ",(int)(nr_int == nr_searched));
+                //[minorScanner scanUpToString:searchedString intoString:&str1];
+                [minorScanner scanUpToString:@"\n" intoString:&str1];
+                NSLog(@"%@",str1);
+                //a = 1;
                 
                 
             }
-            [theScanner scanUpToString:@"\n" intoString:&str1];
             
         }
-        if (a==2)
-        {
-            a=0;
-            break;
-        }
-        //NSLog(@"%@",line);
+        
+        
+        
         
         
     }
-
 }
 
 -(void)ShowAllData
@@ -183,11 +183,32 @@ -(void)WriteOtherWord
     
 }
 
+
+-(NSString *)WaitForCommand
+{
+    //char tab[80];
+    //sscanf("%s",tab);
+    //printf(" printing %s",tab);
+    //NSString *UserWord = [ [NSString alloc] initWithCString:tab encoding:typeUTF8Text];
+    
+    
+    return [[[NSString alloc] initWithData:[[NSFileHandle fileHandleWithStandardInput] availableData] encoding:NSASCIIStringEncoding] stringByTrimmingCharactersInSet:[NSCharacterSet newlineCharacterSet]];
+
+    //return UserWord;
+    
+}
+
 @end
 
 
 
 int main(int argc, const char * argv[]) {
+    NSInteger *nr_int;
+    NSString *str1;
+    //NSInteger nr_searched = [searchedString integerValue];
+    //NSScanner *minorScanner;
+    NSCharacterSet *numbers = [NSCharacterSet characterSetWithCharactersInString:@"1234567890"];
+
     
     
     @autoreleasepool {
@@ -207,17 +228,12 @@ int main(int argc, const char * argv[]) {
         
         NSLog(@" arguments %d ",argc);
         
-        for(int i=0; i< argc; i++)
-        {
-            printf("string: %s \n",argv[i]);
-        }
-        
-        
         
         if (argc<2)
+        {
             //i have to exit
             return 0;
-        
+        }
         
         fileName = [NSString stringWithUTF8String:"adventure.text"];
         //NSLog(@" arg %@ ",  [NSString stringWithFormat:@"%s",argv[1]]);
@@ -231,8 +247,9 @@ int main(int argc, const char * argv[]) {
                                                encoding:NSASCIIStringEncoding
                                                   error:NULL];
         
-        loc1.AllData =[zStr componentsSeparatedByString:@"\n"];
-        //zStr;
+        loc1.AllData =zStr;
+        //[zStr componentsSeparatedByString:@"\n"];
+        
         
         
         //NSString *str1;
@@ -251,50 +268,53 @@ int main(int argc, const char * argv[]) {
         loc1.location_number = [f numberFromString:searchedString];
 
 //        loc1.location_number = [NSNumber numbe];
+        
         [loc1 findExits];
-        [loc1 ShowDescription];
-        return 0;
         
-        NSInteger nr_int;
-        NSString *str1;
-        NSInteger nr_searched = [searchedString integerValue];
-        NSScanner *minorScanner;
-        theScanner = [NSScanner scannerWithString:zStr ];
-        while( [theScanner isAtEnd] == NO)
+        NSScanner *mainScanner;
+        NSString *word = @"";
+        NSString *str2 ;
+        while(![word  isEqual: @"exit"])
         {
-            [theScanner scanUpToString:@"\n" intoString:&str1];
-            minorScanner = [NSScanner scannerWithString:str1];
-            if ([minorScanner scanString:@"-1" intoString:NULL])
-            {
-                NSLog(@" Now next set of infos");
-            }
+            [loc1 ShowDescription];
+            word = [loc1 WaitForCommand];
+            NSLog(@"new => %@",word);
             
-            //if ([minorScanner scanString:searchedString intoString:NULL] )
-            if ( ([minorScanner scanInteger:&nr_int]) && ([ minorScanner scanLocation]<4))
-            {
-                //[minorScanner scanInteger:&nr_int];
-                //NSLog(@" number searching %d %d",(int)nr_int,(int)nr_searched);
-                
-                if (nr_int  == nr_searched)
+            
+            mainScanner = [NSScanner scannerWithString:word ];
+            [mainScanner scanUpToString:@"\n" intoString:&str2];
+            //while( ![mainScanner isAtEnd])
+            //{
+                if ( [ mainScanner scanString:@"dupa " intoString:&str2])
                 {
-                    //NSLog(@" I have find  %@ in %@ ",searchedString, str1);
-                    
-                    //NSLog(@" comparing %d ",(int)(nr_int == nr_searched));
-                    //[minorScanner scanUpToString:searchedString intoString:&str1];
-                    [minorScanner scanUpToString:@"\n" intoString:&str1];
-                    NSLog(str1);
-                    
-                    
+                    NSLog(@" %@ this is very bad word ",str2);
                 }
-                else
-                {                    }
-                //if ([theScanner scanString:searchedString intoString:NULL])
-                //   {
-                
-                // }
                 
-            }
+                if ( [ mainScanner scanString:@"go" intoString:&str2])
+                {
+                    mainScanner = [NSScanner scannerWithString:str2 ];
+                    [mainScanner scanUpToCharactersFromSet:numbers intoString:&str2 ];
+                    NSLog(@" string %@",str2);
+                    if ([mainScanner scanInteger:nr_int])
+                    {
+                        
+                        NSLog(@" go => %ld",(long )nr_int );
+                        
+                        if (nr_int)
+                        {
+                            
+                            loc1.location_number = [NSNumber numberWithInt:nr_int];
+                            NSLog(@" Number location %@",loc1.location_number);
+                        }
+                    }
+                }
+                //[mainScanner setScanLocation:(mainScanner.scanLocation)++ ];
+            //}
+            //loc1.location_number = [f numberFromString:searchedString];
+            
         }
-    }
+        
+        
     return 0;
+    }
 }
